pkg/blockstorage/azure/auth.go

```diff
@@ -0,0 +1,94 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/kanisterio/kanister/pkg/blockstorage"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// currently avaialble types: https://docs.microsoft.com/en-us/azure/developer/go/azure-sdk-authorization
+// to be available with azidentity: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#readme-credential-types
+// to faciliate future migration, only the common ones between two SDKs are listed
+const (
+	CredTypeManagedIdentity   string = "cred-type-msi"
+	CredTypeClientSecret      string = "cred-type-client-secret"
+	CredTypeClientCertificate string = "cred-type-client-certificate"
+)
+
+var availableCredsType = sets.NewString(CredTypeManagedIdentity)
```

Comment:
Why does this have only the type for ManagedIdentity ?  I noticed that you have a function named getClientCredsAuthorizer which gives the impression that we do support something other than ManagedIdentity. But then  this function isCredTypeSupported only returns true if the type is CredTypeManagedIdentity .

Reply:
Because the interface **Authenticated()** only supported ManagedIdentity at that point and that was PR's intention. However, that would be too confusing and thus I had added `ClientCredsAuthorizer`